core/src/main/java/com/linecorp/armeria/common/metric/MetricKeyFunction.java

```diff
@@ -0,0 +1,205 @@
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.server.PathMapping;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+
+/**
+ * Creates a {@link MetricKey} from a {@link RequestLog}.
+ *
+ * @see MetricCollectingClient
+ * @see MetricCollectingService
+ */
+@FunctionalInterface
+public interface MetricKeyFunction {
+
+    /**
+     * Returns the default function that creates a {@link MetricKey} with the specified name parts and
+     * the labels derived from the current {@link PathMapping} (if available) and HTTP (or RPC) method name.
+     * e.g.
+     * <ul>
+     *   <li>Name: {@code ["<namePart[0]>", ... "<namePart[n-1]"]}</li>
+     *   <li>Labels: {@code {"pathMapping=exact:/service/path", "method=POST"}}</li>
+     * </ul>
+     */
+    static MetricKeyFunction ofDefault(String... nameParts) {
+        final List<String> namePartsCopy =
+                validateNameParts(ImmutableList.copyOf(requireNonNull(nameParts, "nameParts")));
+
+        return log -> {
+            final RequestContext ctx = log.context();
+            final Object requestEnvelope = log.requestHeaders();
+            final Object requestContent = log.requestContent();
+
+            String methodName = null;
+
+            if (requestEnvelope instanceof HttpHeaders) {
+                methodName = ((HttpHeaders) requestEnvelope).method().name();
+            }
+
+            if (requestContent instanceof RpcRequest) {
+                methodName = ((RpcRequest) requestContent).method();
+            }
+
+            if (methodName == null) {
+                methodName = MoreObjects.firstNonNull(log.method().name(), "__UNKNOWN_METHOD__");
+            }
+
+            if (ctx instanceof ServiceRequestContext) {
+                final ServiceRequestContext sCtx = (ServiceRequestContext) ctx;
+                final String pathMapping = String.join(",", sCtx.pathMapping().metricName());
+                return new MetricKey(namePartsCopy,
+                                     ImmutableMap.of(BuiltInMetricLabel.method, methodName,
+                                                     BuiltInMetricLabel.pathMapping, pathMapping));
+            } else {
+                return new MetricKey(namePartsCopy,
+                                     ImmutableMap.of(BuiltInMetricLabel.method, methodName));
+            }
+        };
+    }
+
+    /**
+     * Returns the default function that creates a label-less {@link MetricKey} whose name starts with
+     * the specified name parts and ends with the current {@link PathMapping#metricName()} (if available) and
+     * HTTP (or RPC) method name.
+     * e.g. {@code ["<namePart[0]>", ... "<namePart[n-1]", "exact:/service/path", "POST"]}
+     *
+     * @param nameParts the prefix of the created {@link MetricKey}
+     */
+    static MetricKeyFunction ofLabellessDefault(String... nameParts) {
```

Comment:
Maybe just `ofLabelless` or `withoutLabels` as usually there shouldn't be two defaults.